test/test_enzyme.jl

```diff
@@ -1,13 +1,78 @@
 using Oceananigans
+using Oceananigans.TurbulenceClosures: with_tracers
+using Oceananigans.BoundaryConditions: fill_halo_regions!
+using Oceananigans.Fields: ConstantField
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: tracernames
 using Enzyme
 
 # Required presently
 Enzyme.API.runtimeActivity!(true)
-
-EnzymeRules.inactive_type(::Type{<:Oceananigans.Grids.AbstractGrid}) = true
+Enzyme.API.looseTypeAnalysis!(true)
+Enzyme.EnzymeRules.inactive_type(::Type{<:Oceananigans.Grids.AbstractGrid}) = true
+Enzyme.EnzymeRules.inactive_type(::Type{<:Oceananigans.Clock}) = true
+Enzyme.EnzymeRules.inactive_noinl(::typeof(Core._compute_sparams), args...) = nothing
 
 f(grid) = CenterField(grid)
 
+const maximum_diffusivity = 100
+
+"""
+    set_diffusivity!(model, diffusivity)
+
+Change diffusivity of model to `diffusivity`.
+"""
+function set_diffusivity!(model, diffusivity)
+    closure = VerticalScalarDiffusivity(; κ=diffusivity)
+    names = tuple(:c) # tracernames(model.tracers)
+    closure = with_tracers(names, closure)
+    model.closure = closure
+    return nothing
+end
+
+function set_initial_condition!(model, amplitude)
+    # Set initial condition
+    amplitude = Ref(amplitude)
+
+    # This has a "width" of 0.1
+    cᵢ(x, y, z) = amplitude[] * exp(-z^2 / 0.02 - (x^2 + y^2) / 0.05)
+    set!(model, c=cᵢ)
+
+    return nothing
+end
+
+function stable_diffusion!(model, amplitude, diffusivity)
+    set_diffusivity!(model, diffusivity)
+    set_initial_condition!(model, amplitude)
+    
+    # Do time-stepping
+    Nx, Ny, Nz = size(model.grid)
+    κ_max = maximum_diffusivity
+    Δz = 2π / Nz
```

Comment:
why is this the spacing??

```suggestion
    Δz = minimum_zspacing(grid)
```

Reply:
@jlk9 the minimum spacing function is presently nondifferentiable, causing failure. Can you revert this?

[it apparently leads to a call to an eval-like function `ijl_toplevel_eval_in`]

Reply:
Probably the best approach is to make the time-step `Δt` an input into this function (eg pre-calculating it prior to taking the gradient). This is probably what we'd want to do in applications too.

Reply:
OK, then just use the correct spacing `Δz = Lz / Nz`

Reply:
> @jlk9 the minimum spacing function is presently nondifferentiable, causing failure. Can you revert this?
 
OK, then just use the correct spacing `Δz = Lz / Nz`